brainglobe_template_builder/preproc/source_to_raw.py

```diff
@@ -0,0 +1,247 @@
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+from brainglobe_space import AnatomicalSpace
+from brainglobe_utils.IO.image.load import load_any, read_with_dask
+from brainglobe_utils.IO.image.save import save_as_asr_nii
+
+from brainglobe_template_builder.plots import plot_orthographic
+from brainglobe_template_builder.preproc.transform_utils import (
+    downsample_anisotropic_stack_to_isotropic,
+)
+from brainglobe_template_builder.validate import validate_input_csv
+
+
+def _get_subject_path(
+    raw_dir: Path, subject_id: str, output_vox_size: float, mask: bool = False
+) -> Path:
+    """Get path to standardised nifti file for subject_id, and create any
+    required parent directories."""
+
+    # round output vox sizes to nearest int (we don't want decimal
+    # points in the filename)
+    rounded_size = round(output_vox_size)
+    resolution_string = f"res-{rounded_size}x{rounded_size}x{rounded_size}um"
+
+    subject_dir = raw_dir / f"sub-{subject_id}"
+    file_name = f"sub-{subject_id}_{resolution_string}"
+    if mask:
+        dest_path = subject_dir / f"{file_name}_mask_origin-asr.nii.gz"
+    else:
+        dest_path = subject_dir / f"{file_name}_origin-asr.nii.gz"
+
+    dest_path.parent.mkdir(parents=True, exist_ok=True)
+    return dest_path
+
+
+def _process_image(
+    image_path: Path,
+    output_path: Path,
+    origin: str,
+    input_vox_sizes: list[float],
+    output_vox_size: float,
+    mask: bool = False,
+) -> None:
+    """
+    Standardise a single source image.
+
+    Converts to ASR orientation, and downsamples to an isotropic
+    resolution of [output_vox_size, output_vox_size, output_vox_size].
+    Also saves QC plots for this image, to verify ASR orientation.
+
+    Parameters
+    ----------
+    image_path : Path
+        Path to image to standardise.
+    output_path : Path
+        Path to write processed image to.
+    origin : str
+        3-letter anatomical orientation code (e.g., PSL, LSP, RAS)
+    input_vox_sizes : list[float]
+        Input voxel sizes in microns - xyz order.
```

Comment:
```suggestion
        Input voxel sizes in microns - matching anatomical axis order.
```
We are slowly moving away from any xyz terminology where we can, and try to rely on axis numbering (0,1,2) of the Python arrays, which is also implied by our orientation code (ie if the array is PSL, axis 0 points towards posterior).